server/src/main/java/io/crate/planner/optimizer/symbol/Optimizer.java

```diff
@@ -154,6 +155,11 @@ public Symbol visitFunction(io.crate.expression.symbol.Function symbol, Void con
             return sym;
         }
 
+        @Override
+        public Symbol visitAlias(AliasSymbol aliasSymbol, Void context) {
+            return aliasSymbol.symbol().accept(this, context);
+        }
+
```

Comment:
Afaik unwrapping it here instead of doing it as a rule has the downside that rules still might not match, even if they would after unwrapping the alias?

Reply:
I see, good point. 

If I understand correctly, `tryApplyRules` happens in `visitFunction` which is before `visitAlias` happening in `super.visitFunction->twoArgs()`.

I did a change to resolve alias before applying rules - I know it could have been another rule (first element in `Optimzer.optimizeCasts` list.of) but writing a pattern was a bit ugly because of expressing 
`arg(0) == alias || arg(0) == func && func.arg(0) == alias ` - I think either way relatively similar so I went with a simple way